lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8265,6 +8266,19 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
   };
   importAttrsFromDecl(ClangDecl);
 
+  // Unannotated C++ types are considered unsafe.
+  // TODO: add logic for safe cases like aggregates of escapable types.
+  if (const auto *record = dyn_cast<clang::CXXRecordDecl>(ClangDecl)) {
```

Comment:
I don't think we should tie this behavior to C++ mode. All the same checks and heuristics should work equally well whether we're importing as C or as C++, and we try to keep the Swift code working the same way in both modes wherever possible.